### ISSUE TITLE
BrowserTestBaseDefaultThemeRule false positives needs to account for profile

### ISSUE THREAD

**Issue by alexpott** [mglaman/phpstan-drupal#198]
The rule BrowserTestBaseDefaultThemeRule results in false positives. BrowserTestBase::$defaultTheme is not always required. See \Drupal\Core\Test\FunctionalTestSetupTrait::installDefaultThemeFromClassProperty(). If the test is using a profile that uses config installation then the theme is set from that. 

I'm not sure that PHPStan can ever correctly identify this.

**Comment by mglaman** [comment]
That is correct. It reports false positives if there is a profile that provides a default theme. I opted to add "as is" and be loud in this case to try and find an appropriate fix.

One thought I had was to check if the profile is set to a "known" profile that supplies a theme. Like Standard, Umami, whatever test profiles in Drupal core. That would cut down some amount of noise.

@alexpott do you think that'd work? If it's null, check the profile property. If it's in an allowed list, do not report an error.

**Comment by mglaman** [comment]
It also errors on the base class itself.

```
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------
  Line   core/tests/Drupal/Tests/BrowserTestBase.php
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------
  47     Drupal\Tests\BrowserTestBase::$defaultTheme is required. See https://www.drupal.org/node/3083055, which includes recommendations on which theme to
         use.
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------
```

**Comment by mglaman** [comment]
Actually, any abstract class.

```
16:57:05  ------ --------------------------------------------------------------------- 
16:57:05   Line   core/modules/views/tests/src/Functional/ViewTestBase.php             
16:57:05  ------ --------------------------------------------------------------------- 
16:57:05   21     Drupal\Tests\BrowserTestBase::$defaultTheme is required. See         
16:57:05          https://www.drupal.org/node/3083055, which includes recommendations  
16:57:05          on which theme to use.                                               
16:57:05  ------ --------------------------------------------------------------------- 
```

**Comment by mglaman** [comment]
@alexpott off the top of your head, is there a good test in core to analyse for testing?

**Comment by alexpott** [comment]
@mglaman \Drupal\FunctionalTests\Installer\InstallerExistingConfigMultilingualTest has not set \Drupal\Tests\BrowserTestBase::$defaultTheme because it does not need to.

**Comment by alexpott** [comment]
I think the rule should be: ignore this rule if the `\Drupal\Tests\BrowserTestBase::$profile` has a `config/sync/system.theme.yml` or a `config/install/system.theme.yml` file.

**Comment by mglaman** [comment]
@alexpott that _might_ be hard. Don't all profiles have a theme? So maybe if `$profile` is not `minimimal`? 😬

**Comment by alexpott** [comment]
Very few profiles ship with a `config/install/system.theme.yml` or `config/sync/system.theme.yml`. This is not about the theme set in an info.yml file for a profile this is about whether the profile ships with the theme defined in config.

**Comment by alexpott** [comment]
Well I was wrong about `config/install/system.theme.yml` - standard ships with that. But also if you are testing standard then there is no reason to set a default theme. Is there no way to check the profile for the presence of these files. The alternative is to only produce this warning if you are using the default profile for tests which is `testing`.

**Comment by mglaman** [comment]
That seems fair – if `$profile` is `testing`, display the error. Should cover 90% of the use cases. I'll review any of the other testing profiles as well.

**Comment by mglaman** [comment]
I'm checking if the profile belongs to this list:

```
        $testingProfilesWithoutThemes = [
            'testing',
            'nightwatch_testing',
            'testing_config_overrides',
            'testing_missing_dependencies',
            'testing_multilingual',
            'testing_multilingual_with_english',
            'testing_requirements',
        ];
```

That _should_ cover everything.

**Comment by jibran** [comment]
Created #234 for upgrade path tests.
